tikv-client/src/main/java/com/pingcap/tikv/jdbc/JDBCClient.java

```diff
@@ -38,57 +53,19 @@ public JDBCClient(String url, Properties properties) {
   }
 
   public List<String> showGrants() {
-    try (Connection connection = driver.connect(url, properties);
-        Statement tidbStmt = connection.createStatement();
-        ResultSet resultSet = tidbStmt.executeQuery(SQL_SHOW_GRANTS)) {
-
-      List<String> result = new ArrayList<>();
-      ResultSetMetaData rsMetaData = resultSet.getMetaData();
-
-      while (resultSet.next()) {
-        for (int i = 1; i <= rsMetaData.getColumnCount(); i++) {
-          result.add(resultSet.getString(i));
-        }
-      }
-
-      return result;
+    try {
+      return query(SQL_SHOW_GRANTS, (rs, rowNum) -> rs.getString(1));
     } catch (SQLException e) {
       return Collections.emptyList();
     }
   }
 
   public String getPDAddress() throws SQLException {
-    try (Connection connection = driver.connect(url, properties);
-        Statement tidbStmt = connection.createStatement();
-        ResultSet resultSet = tidbStmt.executeQuery(GET_PD_ADDRESS)) {
-
-      List<String> result = new ArrayList<>();
-      ResultSetMetaData rsMetaData = resultSet.getMetaData();
-
-      while (resultSet.next()) {
-        for (int i = 1; i <= rsMetaData.getColumnCount(); i++) {
-          result.add(resultSet.getString(i));
-        }
-      }
-      return result.get(0);
-    }
+    return queryForObject(GET_PD_ADDRESS, (rs, rowNum) -> rs.getString("INSTANCE"));
   }
 
   public String getCurrentUser() throws SQLException {
-    try (Connection connection = driver.connect(url, properties);
-        Statement tidbStmt = connection.createStatement();
-        ResultSet resultSet = tidbStmt.executeQuery(SELECT_CURRENT_USER)) {
-
-      List<String> result = new ArrayList<>();
-      ResultSetMetaData rsMetaData = resultSet.getMetaData();
-
-      while (resultSet.next()) {
-        for (int i = 1; i <= rsMetaData.getColumnCount(); i++) {
-          result.add(resultSet.getString(i));
-        }
-      }
-      return result.get(0);
-    }
+    return queryForObject(SELECT_CURRENT_USER, (rs, rowNum) -> rs.getString("current_user()"));
```

Comment:
hm... Not sure if column name is case sensitive or not. You can actually use rs.getString(1) since you already know there is only one column and that's the only one column you want to read.

Reply:
ACK